BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -80,17 +91,17 @@ public void Convert()
 					case "target_give":
 						ignoreEntity = true;
 						break;
-					default:
-						{
-							if (entity.ClassName.StartsWith("weapon_"))
+						default:
+							{
+							if ((entity.ClassName.StartsWith("weapon_")) && (entity.Name == ""))
```

Comment:
Why are you checking if the targetname is blank? Is this meant to prevent weapons from getting converted before converting the give triggers? If so, you should revert this change and we should implement a proper fix instead.

Reply:
I should also note that this will break maps with weapons that have a targetname assigned but no linked target_give.

Reply:
do we not want to only convert weapon_ entities that aren't targets for this part?